Edits for punctuation, usage, and clarity (#335)
mondoohq/mondoo-operator: docs/user-manual.md

# User manual
This user manual describes how to install and use the Mondoo operator.

## Mondoo Operator Installation
Install the Mondoo operator using kubectl, Helm, or Operator Lifecycle Manager.

### Installing with kubectl
Follow this step to set up the Mondoo operator using kubectl and a manifest file.

Precondition: kubectl with cluster admin access

To install with kubectl, apply the operator manifests:

```bash
kubectl apply -f https://github.com/mondoohq/mondoo-operator/releases/latest/download/mondoo-operator-manifests.yaml
```
    
or

```bash
curl -sSL https://github.com/mondoohq/mondoo-operator/releases/latest/download/mondoo-operator-manifests.yaml > mondoo-operator-manifests.yaml
kubectl apply -f mondoo-operator-manifests.yaml
```

### Installing with Helm

> Follow these steps to set up a development Kubernetes to test the operator using [Helm](https://helm.sh/).

Reviewer: ```suggestion
Follow these steps to set up the Mondoo Operator using [Helm](https://helm.sh/).
```

This user manual describes how to install and use the Mondoo operator.

## Mondoo Operator Installation
Install the Mondoo operator using kubectl, Helm, or Operator Lifecycle Manager.

### Installing with kubectl
Follow this step to set up the Mondoo operator using kubectl and a manifest file.

Precondition: kubectl with cluster admin access

&rarr; Apply the operator manifests:

>     ```bash

Author: I fixed the markdown by removing the indent. Thanks for catching that!

    }
    ```

3. Store the service account as a secret in the Mondoo namespace:
    ```bash
    kubectl create secret generic mondoo-client --namespace mondoo-operator --from-file=config=creds.json
    ```

> ## Creating MondooAuditConfig

Reviewer: MondooAuditConfig is a type of object (of which there can be more than one). In the same way we would say "first create a Namespace" or "update the contents of the Secret", it makes sense to me to talk about "creating a MondooAuditConfig".
```suggestion
## Creating a MondooAuditConfig
```

        enable: true
    ```

2. Apply the configuration:
    ```bash
    kubectl apply -f mondoo-config.yaml
    ```

## Deploying the admission controller
Kubernetes webhooks require TLS certs to establish the trust between the certificate authority listed in  `ValidatingWebhookConfiguration.Webhooks[].ClientConfig.CABundle` and the TLS certificates presented when connecting to the HTTPS endpoint specified in the webhook.

You can choose one of three approaches:
- Install and use cert-manager to automate the creation and update of the TLS certs
- Use the OpenShift certificate creation/rotation features
- Create (and rotate) your own TLS certificates manually

A working setup shows the webhook pod processing the created/modified/deleted pods.

Display the logs in one window:
```bash

> kubectl logs -f deployment/mondoo-operator/mondoo-operator-webhook-manager -n mondoo-operator

Reviewer: The names of objects have subtly changed recently
```suggestion
kubectl logs -f deployment/mondoo-client-webhook-manager -n mondoo-operator
```

```

> And create/modify/delete a pod in another window:

Reviewer: `Pod` is the proper name of a specific type of object in k8s. In the same way that we refer to Deployments and ConfigMaps and Secrets, a Pod is also one of these "special named objects". See the k8s docs: https://kubernetes.io/docs/concepts/workloads/pods/

Author: I'm reading their docs... their choices strike me as inconsistent. But I agree we should follow them.

          mode: cert-manager
    ```

> The admission controller `Deployment` should start. `ValidatingWebhookConfiguration` should be annotated to insert the certificate authority data. cert-manager creates a secret named `webhook-serving-cert` that contains the TLS certificates.

Reviewer: ```suggestion
The admission controller `Deployment` should start.  The`ValidatingWebhookConfiguration` should be annotated to insert the certificate authority data. cert-manager creates a secret named `webhook-serving-cert` that contains the TLS certificates.
```

### Manually creating TLS certificates using OpenSSL
You can manually create the TLS certificate required for the admission controller. These steps show one method:

1. Create a key for your certificate authority:
   ```bash
   openssl genrsa -out ca.key 2048
   ```

2. Generate a certificate for your certificate authority:
   ```bash
   openssl req -x509 -new -nodes -key ca.key -subj "/CN=Webhook Issuer" -days 10000 -out ca.crt
   ```

> 3. Generate a key for the webhook server pod:

Reviewer: ```suggestion
3. Generate a key for the webhook server Pod:
```

    subjectAltName=@alt_names
    ```

5. Generate the certificate signing request:
   ```bash
   openssl req -new -key server.key -out server.csr -config csr.conf
   ```

6. Generate the certificate for the webhook server:
   ```bash
   openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -days 10000 -extensions v3_ext -extfile csr.conf
   ```

> 7. Create the secret holding the TLS certificate data:

Reviewer: A Secret is the proper name of a specific type of k8s resource/object. https://kubernetes.io/docs/concepts/configuration/secret/
```suggestion
7. Create the Secret holding the TLS certificate data:
```
